app/controllers/locations_controller.rb

```diff
@@ -15,7 +15,165 @@ def index
   end
 
   def new
-    @location = Location.new
+    @location = Location.new(scheme: @scheme)
+    @location.save!
+    redirect_to scheme_location_postcode_path(@scheme, @location, route: params[:route])
+  end
```

Comment:
```suggestion
  def create
    @location = @scheme.locations.create!
    redirect_to scheme_location_postcode_path(@scheme, @location, route: params[:route])
  end
```

It's very usual to have `new` create a resource. Are you sure you don't want to `POST /locations` ?